src/acl/AtStep.cc

```diff
@@ -8,25 +8,29 @@
 
 #include "squid.h"
 
-#if USE_OPENSSL
-
 #include "acl/AtStep.h"
 #include "acl/AtStepData.h"
 #include "acl/FilledChecklist.h"
 #include "client_side.h"
 #include "http/Stream.h"
+#if USE_OPENSSL
 #include "ssl/ServerBump.h"
+#endif
 
 int
-ACLAtStepStrategy::match (ACLData<Ssl::BumpStep> * &data, ACLFilledChecklist *checklist)
+ACLAtStepStrategy::match (ACLData<XactionSteps> * &data, ACLFilledChecklist *checklist)
 {
+    Must(checklist->request);
```

Comment:
Are you sure it is OK to throw here? Should not we return a mismatch instead? Same for the `checklist->request->masterXaction` Must() below.

Reply:
Yes it was wrong.

Fixed with commit 8f212b54d6